### PR TITLE
Have udp tests return idiomatic ginkgo errors, update reporting of mutating parts

### LINKED ISSUE #95216: weird output, Networking Granular Checks: Services should update nodePort: udp [Slow]
**What happened**:

final passing output: 
```
ep 30 23:44:22.435: INFO: Failed to execute "echo hostName | nc -w 1 -u 12.10.10.67 32362 | grep -v '^\\s*$'": command terminated with exit code 1, stdout: "", stderr: ""
Sep 30 23:44:22.435: INFO: Found all expected endpoints: []
```

**What you expected to happen**:

Probably if this test passes, it should be showing several real endpoints.  I don't know exactly what it does, but im assuming that the output could be clarified a little. 

**How to reproduce it (as minimally and precisely as possible)**:

run `sonobuoy run --e2e-focus="Services should update nodePort: udp"` and look at the logs.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.20 (sonobuoy)

### PR DESCRIPTION
**Pull request by jayunit100**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

fixes #95216 
- by adding `ginkgo.By` directives to explain the mutations that are happening
- print the number of expected services out in the verification. 

**Which issue(s) this PR fixes**:

Fixes #95216

**Does this PR introduce a user-facing change?**:

-->
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/network/utils.go [hunk 1]
```diff
@@ -418,8 +421,8 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
 
 		// Check against i+1 so we exit if minTries == maxTries.
 		if eps.Equal(expectedEps) && i+1 >= minTries {
-			framework.Logf("Found all expected endpoints: %+v", eps.List())
-			return
+			framework.Logf("Found all %+v expected endpoints: %+v", len(eps.List()), eps.List())
```
Comment (jayunit100) [inline test/e2e/framework/network/utils.go:424]: added this bc in large ci runs being able to see the number makes it obvious to see that this function, for example, is being called to validate 0 endpoints (imo there's maybe a better way then calling it w/ eps=[] but we can save that debate for later date :)), for now just make the logging clear

**Comment by jayunit100** [comment]
@aojea 2nd part of https://github.com/kubernetes/kubernetes/pull/93837

**Comment by aojea** [comment]
Ccan you submit a new PR for the changes to `config.DialFromNode()` , seems unrelated to this PR.
Also, related to that, there are more places using the method that need to be changed
```
[aojea@aojea-laptop kubernetes]$ grep -nr DialFromNode test/
test/e2e/network/networking.go:213:			config.DialFromNode("http", config.ClusterIP, e2enetwork.ClusterHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:215:			config.DialFromNode("http", config.NodeIP, config.NodeHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:221:			config.DialFromNode("udp", config.ClusterIP, e2enetwork.ClusterUDPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:224:			config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:339:			config.DialFromNode("http", config.NodeIP, config.NodeHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:344:			config.DialFromNode("http", config.NodeIP, config.NodeHTTPPort, config.MaxTries, config.MaxTries, sets.NewString())
test/e2e/network/networking.go:351:			config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:356:			config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, config.MaxTries, sets.NewString())
test/e2e/framework/network/utils.go:381:// DialFromNode executes a tcp or udp request based on protocol via kubectl exec
test/e2e/framework/network/utils.go:391:func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targetPort, maxTries, minTries int, expectedEps sets.String) {
test/e2e/common/networking.go:106:				config.DialFromNode("http", endpointPod.Status.PodIP, e2enetwork.EndpointHTTPPort, config.MaxTries, 0, sets.NewString(endpointPod.Name))
test/e2e/common/networking.go:120:				config.DialFromNode("udp", endpointPod.Status.PodIP, e2enetwork.EndpointUDPPort, config.MaxTries, 0, sets.NewString(endpointPod.Name))
```

**Comment by jayunit100** [comment]
> [aojea@aojea-laptop kubernetes]$ grep -nr DialFromNode test/
ha other commit got lost in translation, :) didn't know you'd review it so quickly . pushed :)

**Comment by jayunit100** [comment]
I think they're all the same PR ? 

1) `DialFromNode` needs to return an error so the comments need updating to describe the new method signature
2) All the other tests need to now process the new DialFromNode output by failing.

File: test/e2e/framework/network/utils.go [hunk 2]
```diff
@@ -424,8 +427,8 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
...
```
Comment (aojea) [inline test/e2e/framework/network/utils.go:430]:
```suggestion
			framework.Logf("Found all %d expected endpoints: %+v", eps.Len(), eps.List())
```

**Comment by aojea** [comment]
> test/e2e/framework/network/utils.go:441:20: error strings should not be capitalized or end with punctuation or a newline

**Comment by jayunit100** [comment]
updated thx @aojea

**Review by spiffxp — COMMENTED** [review]
I'm in favor of propagating the errors but I have a nit about the comment changes

File: test/e2e/framework/network/utils.go [hunk 3]
```diff
@@ -380,17 +380,20 @@ func (config *NetworkingTestConfig) GetHTTPCodeFromTestContainer(path, targetIP
...
+//   success. If 0, then we return as soon as all endpoints succeed.
+// - There is no logical change to test results if faillures happen AFTER endpoints have succeeded,
+//   hence over-padding minTries will NOT reverse a successful result and is thus not very useful yet
+//   (See the TODO about checking probability, which isnt implemented yet).
+// - maxTries is the maximum number of curl/echo attempts before an error is returned.  The
+//   smaller this number is, the less 'slack' there is for declaring success.
+// - if maxTries < expectedEps, this test is guaranteed to return an error, because all endpoints wont be hit.
+// - maxTries == minTries will return as soon as all endpoints succeed (or fail once maxTries is reached without
+//   success on all endpoints).
+//   In general its prudent to have a high enough level of minTries to guarantee that all pods get a fair chance at receiving traffic.
```
Comment (spiffxp) [inline test/e2e/framework/network/utils.go:395]: Given all the comment space that is spent on minTries/maxTries, would it be prudent to log those in test output?  Right now only maxTries is logged, and only on error

Comment (jayunit100) [reply test/e2e/framework/network/utils.go:395]: fixeddddd, added a log message below

File: test/e2e/framework/network/utils.go [hunk 4]
```diff
@@ -380,17 +380,20 @@ func (config *NetworkingTestConfig) GetHTTPCodeFromTestContainer(path, targetIP
 	return code, nil
 }
 
-// DialFromNode executes a tcp or udp request based on protocol via kubectl exec
+// DialFromNode executes a tcp or udp curl/echo request based on protocol via kubectl exec
```
[the remark this answers is not in the document]
Comment (jayunit100) [reply test/e2e/framework/network/utils.go:383]: +1 will add nc

**Comment by jayunit100** [comment]
ok updated @spiffxp

File: test/e2e/network/networking.go [hunk 5]
```diff
@@ -321,39 +333,53 @@ var _ = SIGDescribe("Networking", func() {
...
 
+			ginkgo.By("Deleting the node port access point")
 			config.DeleteNodePortService()
 
-			ginkgo.By(fmt.Sprintf("dialing(udp) %v (node) --> %v:%v (nodeIP)", config.NodeIP, config.NodeIP, config.NodeUDPPort))
-			config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, config.MaxTries, sets.NewString())
+			ginkgo.By(fmt.Sprintf("dialing(udp) %v (node) --> %v:%v (nodeIP) and getting ZERO host endpoints", config.NodeIP, config.NodeIP, config.NodeUDPPort))
+			err = config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, config.MaxTries, sets.NewString())
+			if err != nil {
+				framework.Failf("Failure validating that node port service STOPPED removed properly: %v", err)
```
Comment (aojea) [inline test/e2e/network/networking.go:381]: nit, it is not really stopped, the service is removed, but not a big deal, at least the message is different and will help to spot easier where the test fails

**Comment by aojea** [comment]
/lgtm
thanks

File: test/e2e/network/networking.go (hunk 5 above)
Comment (jayunit100) [reply test/e2e/network/networking.go:381]: ok, will keep for now since its lgtm

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**